### PR TITLE
Cleanup `rustc_features` some more

### PR DESCRIPTION
**Pull request by nnethercote**
The sequel to #116437.

r? @Nilstrieb

### CODE REVIEW

File: compiler/rustc_feature/src/active.rs [hunk 1]
```diff
@@ -148,8 +148,7 @@ macro_rules! declare_features {
 // was set. This is most important for knowing when a particular feature became
 // stable (active).
 //
-// Note that the features are grouped into internal/user-facing and then
```
Comment (Nilstrieb) [inline compiler/rustc_feature/src/active.rs:159]: They are split into several categories, including internal, aren't they?

Comment (nnethercote) [reply compiler/rustc_feature/src/active.rs:159]: They are, but the ordering doesn't take that into account. It used to, but at some point it was changed to be purely alphabetical. So the old comment was wrong.

Comment (Nilstrieb) [reply compiler/rustc_feature/src/active.rs:159]: https://github.com/rust-lang/rust/pull/116550/commits/7764b66368de7b7f2b8572cee70fce91dc8e1e5b#diff-a82bdbaaa3c79096a30755dce8896aaf2885128a7f3f3a290072b767f05b6095L233 this is starting with a in the second group

Comment (nnethercote) [reply compiler/rustc_feature/src/active.rs:159]: Right, my mistake. I have removed that commit.

**Comment by nnethercote** [comment]
I have updated to remove the commit with the comment change.

**Review by Noratrieb — APPROVED** [review]

**Comment by Nilstrieb** [comment]
why do you need to make your access issues everyone's problem, rustfix
@bors retry

**Pull request merged**